Sell/FulfilmentPoint/Tests/FulfilmentPointRepositoryTests.swift

```diff
@@ -0,0 +1,332 @@
+            .GetFulfilmentPointsQuery
+            .Data>(
+            data: data,
+            extensions: nil,
+            errors: nil,
+            source: .cache,
+            dependentKeys: nil)
+        graphQLManager.resultReturns = .success(expectedResult)
+
+        let expectation = XCTestExpectation(description: "callback is fulfilled")
+        sut.getFulfilmentPoints(
+            pageSize: 1,
+            page: 1,
+            filters: nil
+        ) { result in
+            guard case let .success(returnedResponse) = result else {
+                return XCTFail("This test should return success")
+            }
+            XCTAssertEqual(
+                returnedResponse.items.count,
+                expectedResult.data?.getFulfilmentPoints?.edges?.count)
+            XCTAssertEqual(
+                returnedResponse.nextPage,
+                expectedResult.data?.getFulfilmentPoints?.nextPage)
+            expectation.fulfill()
+        }
+        wait(for: [expectation], timeout: 0.01)
+    }
+
+    func test_getFulfilmentPoints_graphQLManagerReturnsError_completeWithFailureCase() {
+        let tuple = makeGraphQLManagerAndSUT(ofType: ApolloType.GetFulfilmentPointsQuery.Data.self)
+        let graphQLManager = tuple.graphQLManager
+        let sut = tuple.sut
+        graphQLManager.resultReturns = .failure(DummyError.failure)
+
+        let expectation = XCTestExpectation(description: "callback is fulfilled")
+        sut.getFulfilmentPoints(
+            pageSize: 1,
+            page: 1,
+            filters: nil
+        ) { result in
+            guard case let .failure(returnedError) = result else {
+                return XCTFail("This test should return failure")
+            }
+            XCTAssertEqual(returnedError as? DummyError, DummyError.failure)
+            expectation.fulfill()
+        }
+        wait(for: [expectation], timeout: 0.01)
+    }
+
+    func test_getFulfilmentPointById_graphQLManagerHasData_completeWithSuccessCase() {
+        let tuple = makeGraphQLManagerAndSUT(ofType: ApolloType.GetFulfilmentPointByIdQuery.Data.self)
+        let graphQLManager = tuple.graphQLManager
+        let sut = tuple.sut
+        let fulfilmentPointFragment = makeFulfilmentPointFragment()
+
+        let getFulfilmentPoint = ApolloType
+            .GetFulfilmentPointByIdQuery
+            .Data
+            .GetFulfilmentPoint(unsafeResultMap: [:])
+        var data = ApolloType
+            .GetFulfilmentPointByIdQuery
+            .Data(getFulfilmentPoint: getFulfilmentPoint)
+        data.getFulfilmentPoint?.fragments.fragmentFulfilmentPoint = fulfilmentPointFragment
+        let expectedResult = GraphQLResult<
+            ApolloType
+            .GetFulfilmentPointByIdQuery
+            .Data>(
+            data: data,
+            extensions: nil,
+            errors: nil,
+            source: .cache,
+            dependentKeys: nil)
+        graphQLManager.resultReturns = .success(expectedResult)
```

Comment:
So much wrapping required to deal with GraphQL's type system. Well done picking it apart.